modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/TableRowConverterImpl.java

```diff
@@ -59,7 +60,16 @@ public class TableRowConverterImpl implements TableRowConverter {
         this.schemaDescriptor = schemaDescriptor;
         this.fullTupleSchema = fullTupleSchema;
 
-        this.skipTrimming = requiredColumns == null;
+        tupleFactory = requiredColumns == null ? () ->
```

Comment:
Let's split into 2 contructors with non-nullable `requiredColumns` and without `requiredColumns` parameter.

Reply:
Or maybe create a child class ProjectableTableRowConverter that accepts `requiredColumns`.
In this case, you could replace a factory with a protected method, which can be overridden in a child class.